dev/io.openliberty.microprofile.telemetry.1.0.internal.container_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/MetricsApiOtelCollectorTest.java

```diff
@@ -0,0 +1,162 @@
+        server.addEnvVar(TestConstants.ENV_OTEL_SERVICE_NAME, "PrometheusOtelCollectorTest");
+        server.addEnvVar(TestConstants.ENV_OTEL_SDK_DISABLED, "false");
+
+        // Construct the test application
+        WebArchive spanTest = ShrinkWrap.create(WebArchive.class, "spanTest.war")
+                                        .addPackage(TestResource.class.getPackage());
+        ShrinkHelper.exportAppToServer(server, spanTest, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer();
+    }
+
+    @Test
+    public void testLongUpDownCounter() throws Exception {
+        HttpRequest request = new HttpRequest(server, "/spanTest/longUpDownCounterCreated");
+        String response = request.run(String.class);
+
+        Log.info(c, "testLongUpDownCounter", "response: " + response);
+        Thread.sleep(3000);
+        getApiMetrics("testLongUpDownCounter", "gauge", "-20");
+    }
+
+    @Test
+    public void testCounter() throws Exception {
+        HttpRequest request = new HttpRequest(server, "/spanTest/longCounterCreated");
+        String response = request.run(String.class);
+        Log.info(c, "testCounter", "response: " + response);
+        Thread.sleep(3000);
+        getApiMetrics("testLongCounter", "counter", "20");
+    }
+
+    @Test
+    public void testLongHistogram() throws Exception {
+        HttpRequest request = new HttpRequest(server, "/spanTest/longHistogramCreated");
+        String response = request.run(String.class);
+        Thread.sleep(3000);
+        Log.info(c, "testLongHistogram", "response: " + response);
+        getApiMetrics("testLongHistogram", "histogram", "20");
+    }
+
+    @Test
+    public void testDoubleCounter() throws Exception {
+        HttpRequest request = new HttpRequest(server, "/spanTest/doubleCounterCreated");
+        String response = request.run(String.class);
+        Thread.sleep(3000);
+        Log.info(c, "testDoubleCounter", "response: " + response);
+        getApiMetrics("testDoubleCounter", "counter", "40");
+    }
+
+    @Test
+    public void testDoubleUpDownCounter() throws Exception {
+        HttpRequest request = new HttpRequest(server, "/spanTest/doubleUpDownCounterCreated");
+        String response = request.run(String.class);
+        Thread.sleep(3000);
+        Log.info(c, "testDoubleUpDownCounter", "response: " + response);
+        getApiMetrics("testDoubleUpDownCounter", "counter", "-40");
+    }
+    /**
+     * Gets metrics from otelcollector:3131/metrics in the prometheus format.
+     * For more info on the Prometheus metrics format: 
+     * https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#text-format-details
+     */
+    public void getApiMetrics(String name, String type, String value) throws Exception {
+        String result = client.dumpMetrics();
```

Comment:
Not the fault of this PR, but does this have anything to protect against the metric url being called before the interval ends? I don't see anything